tests/test_end_to_end.py

```diff
@@ -96,6 +96,34 @@ def test_dataframe_transform(self):
         transformed_df = df.ai.transform("what is the name with oldest age?")
         self.assertEqual(transformed_df.collect()[0][0], "Bob")
 
+    def test_transform_col_query_nondescriptive(self):
+        """Test that agent selects correct query column, even with non-descriptive column names,
+        by using sample column values"""
+        df = self.spark_ai._spark.createDataFrame(
+            [
+                ("Shanghai", 31, "China"),
+                ("Seattle", 30, "United States"),
+                ("Austin", 33, "United States"),
+                ("Paris", 29, "France"),
+            ],
+            ["col1", "col2", "col3"],
+        )
+        transformed_df = df.ai.transform("what city had the warmest temperature?")
+        self.assertEqual(transformed_df.collect()[0][0], "Austin")
+
+    def test_transform_col_query_unintuitive(self):
```

Comment:
This test case seems too much. I am not sure if it works with other LLMs in the future...